packages/verkle/src/verkleTree.ts

```diff
@@ -277,10 +278,23 @@ export class VerkleTree {
     let currentDepth = leafNode.depth
 
     while (currentDepth > 0) {
-      const parentKey = currentKey.slice(0, -1)
       const parentIndex = currentKey[currentKey.length - 1]
-      const parentNode = InternalNode.create(currentDepth, this.verkleCrypto)
-      parentNode.children[parentIndex] = currentNode
+      // Parent key should be at least one shorter than the current key length
+      const parentKey = currentKey.slice(0, currentKey.length - 1)
+      const rawNode = await this.get(parentKey)
+      let parentNode
+      // TODO: Determine how to decide if we need to go further up the tree or create a new internal node here
+      // Currently, this code would insert a new internal node at every step up the trie (which we don't want)
```

Comment:
The way I'm thinking of this, we could find the nearest existing path to the leaf node that we want to insert, while keeping track of all the tries that have led to that, and update them sequentially, rather than coming from the child and looking up each parent. Does that make sense?